test/common/upstream/health_checker_impl_test.cc

```diff
@@ -1249,11 +1249,92 @@ TEST_F(HttpHealthCheckerImplTest, SuccessServiceRegexPatternCheck) {
   EXPECT_EQ(Host::Health::Healthy, cluster_->prioritySet().getMockHostSet(0)->hosts_[0]->health());
 }
 
+TEST_F(HttpHealthCheckerImplTest, SuccessServiceCheckWithCustomHostValueOnTheHost) {
```

Comment:
a comment explaining what is being tested in this and the test below would be useful, the name is quite a handful